service/history/reset/resetter.go

```diff
@@ -202,32 +202,58 @@ func (r *workflowResetterImpl) prepareResetWorkflow(
 		return nil, err
 	}
 
-	// TODO add checking of reset until event ID == decision task started ID + 1
 	decision, ok := resetMutableState.GetInFlightDecision()
-	if !ok || decision.StartedID+1 != resetMutableState.GetNextEventID() {
-		return nil, &shared.BadRequestError{
-			Message: fmt.Sprintf("Can only reset workflow to DecisionTaskStarted + 1: %v", baseRebuildLastEventID+1),
+	if ok {
+		// reset workflow has decision task start
+		if decision.StartedID+1 != resetMutableState.GetNextEventID() {
+			return nil, &shared.BadRequestError{
+				Message: fmt.Sprintf("Can only reset workflow to DecisionTaskStarted + 1: %v", baseRebuildLastEventID+1),
+			}
 		}
-	}
-	if len(resetMutableState.GetPendingChildExecutionInfos()) > 0 {
-		return nil, &shared.BadRequestError{
-			Message: fmt.Sprintf("Can only reset workflow with pending child workflows"),
+		if len(resetMutableState.GetPendingChildExecutionInfos()) > 0 {
+			return nil, &shared.BadRequestError{
+				Message: fmt.Sprintf("Can only reset workflow with pending child workflows"),
```

Comment:
typo: "Can not only reset workflow with pending child workflows"

Reply:
updated